java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -277,6 +277,7 @@ private void reportIssue(Tree tree, SymbolicValue denominator, Symbol denominato
       String expressionName = expression.is(Tree.Kind.IDENTIFIER) ? ("\"" + ((IdentifierTree) expression).name() + "\"") : "this expression";
       List<Class<? extends Constraint>> domains = Collections.singletonList(ZeroConstraint.class);
       Set<List<JavaFileScannerContext.Location>> flows = FlowComputation.flow(context.getNode(), denominator, domains, denominatorSymbol).stream()
+        .filter(f -> !f.isEmpty())
```

Comment:
Can't be done easily, because we rely on the fact that always at least one empty flow is returned, because messages are added to the result.